icu4c/source/i18n/measunit_extra.cpp

```diff
@@ -758,6 +657,43 @@ const char *SingleUnitImpl::getSimpleUnitID() const {
     return gSimpleUnits[index];
 }
 
+void SingleUnitImpl::appendNeutralIdentifier(CharString &result, UErrorCode &status) const {
+    int32_t absPower = std::abs(this->dimensionality);
+
+    U_ASSERT(absPower > 0); // "this function does not support the dimensionless single units";
+
+    if (absPower == 1) {
+        // no-op
+    } else if (absPower == 2) {
+        result.append(StringPiece("square-"), status);
+    } else if (absPower == 3) {
+        result.append(StringPiece("cubic-"), status);
+    } else if (absPower <= 15) {
+        result.append(StringPiece("pow"), status);
+        // TODO: could we use std::to_string
+        result.append(StringPiece(std::to_string(absPower)), status);
```

Comment:
Don't use std::to_string.  Use ICU_Utility::appendNumber instead (feel free to add a version of that helper function that works on 8-bit strings), or carry over the `absPower + '0'` logic from before.

Reply:
Okay, Now I have added an `appendNumber` function for ICU_Utility and for `CharString`, I prefer the one in the CharString.

Choose one in order to finalise the PR :)